framework/src/meshgenerators/MeshGenerator.C

```diff
@@ -140,7 +140,10 @@ MeshGenerator::getMesh(const std::string & param_name, const bool allow_invalid
   const MeshGeneratorName * name = getMeshGeneratorNameFromParam(param_name, allow_invalid);
   if (!name)
     return _null_mesh;
-  return getMeshByName(*name);
+  auto & mesh = getMeshByName(*name);
+  if (!mesh && !allow_invalid)
+    mooseError("Returning a null mesh. The previous mesh generator must not have returned a mesh.");
```

Comment:
the new error is coming from adding this check, not from having changed the RGMB to return null meshes when skipping mesh generation right?

Is it better to allow_invalid for getMesh calls for sub-generators
or better to have a new API meant to only be used for sub-generators

Reply:
@loganharbour

Reply:
These errprs are coming primarily from the fact that getMesh allowed you to pass null meshes by default in the constructor, and so now we explicitly have to call allow_invalid=true when a mesh generator might call getMesh on a mesh that hasn't been generated yet. This pattern occurs not just in subgenerator calls like in RGMB, but also in normal MG's where we are trying to save mesh properties in the constructor and updating them in `generate`, see the test failures in the framework test but basically if we went down this route, this would lead to a lot changes needed to the test suite.

I'm wondering if we should alternatively create a new base class similar to `MeshGenerator` that only allows you to update metadata, however, it isn't clear to me how existing MG's would inherit this base class if this depends on a input parameter being true or not

Reply:
I think Logan is more comfortable with this metadata and has considered the design carefully. But feel free to design a new base class (maybe more of an interface class?) to handle the metadata outside of `MeshGenerator` if you think that can be enabling here and is widely applicable to our current uses as well